### ISSUE TITLE
Providing cross-validation sets for LSTM

### ISSUE THREAD

**Issue by shaimaahegazy** [joncox123/Cortexsys#5]
Is there a way to provide cross-validation/test sets for the LSTM training function? I can understand that the function is defined as follows:

```
gradientDescentAdaDelta(costFunc, nn, defs, Xts, Yts, yts, y, 'Training
Entire Network');

```

My question is what are the : Xts, Yts, yts and  y variables?

Thanks a lot.

**Comment by guanyou** [comment]
From what I know, like for the MNIST example data, X being the training data and y being the testing data,

X = (28_28... image data)= 784 \* 60000 .... variations  training data input
Xts =  (28_28...image data)= 784 \* 10000... variations   test data input
y =(label of image data) 1 \* 60000...variations           training data output
yts =   (label for image data)1 \* 10000 .. .variations    test data output

**Comment by shaimaahegazy** [comment]
Thanks for your reply @guanyou. Do you know where I could get the MNIST data used in the example?

**Comment by guanyou** [comment]
I'm guessing u could get it from below, or just implement those data conversion code found online with the original dataset found here http://yann.lecun.com/exdb/mnist/

http://www.cs.toronto.edu/~norouzi/research/mlh/data/mnist-full.mat
http://cs.nyu.edu/~roweis/data.html

**Comment by guanyou** [comment]
But so far, it only work with ConvNet classifier for me, the rest seems to fail on pretrain/ training with cost: NaN

**Comment by RMano82** [comment]
Hi @shaimaahegazy . I wonder if you have managed to run any of the examples provided with the toolbox where the network converges.

I, so far tried the MNIST_ConvNet_Classifier.m and the MNIST_Deep_Classifier.m where I download the mnist-full.mat from 
http://www.cs.toronto.edu/~norouzi/research/mlh/data/mnist-full.mat
However, the networks in both examples do not converge. 
Any help will be much appreciated
